app/api/src/crud/crud_read_heatmap.py

```diff
@@ -439,37 +430,74 @@ async def read_heatmap(
 
     def sort_and_unique(self, grid_ids, traveltimes):
         sorted_table, unique = {}, {}
-        for op in traveltimes.kesys():
+        for op in traveltimes.keys():
             sorted_table[op], unique[op] = heatmap_core.sort_and_unique_by_grid_ids(
                 grid_ids[op], traveltimes[op]
             )
         return sorted_table, unique
 
-    def do_calculations(self, sorted_table, unique, heatmap_settings):
+    def do_calculations(self, sorted_table, uniques, heatmap_settings):
```

Comment:
We need a logic here where we are grouping the results to the requested resolution. So if the user requests resolution 9 for instance we should groupfrom resolution 10. I would do the grouping simply by building the average and using the H3 library to get the parent.